CalibTracker/SiStripChannelGain/plugins/SiStripGainFromCalibTree.cc

```diff
@@ -684,7 +684,7 @@ void SiStripGainFromCalibTree::algoBeginJob(const edm::EventSetup& iSetup)
 	for(unsigned int i=0;i<Det.size();i++){  //Make two loop such that the Pixel information is added at the end --> make transition simpler
 		DetId  Detid  = Det[i]->geographicalId();
 		int    SubDet = Detid.subdetId();
-		if( SubDet == PixelSubdetector::PixelBarrel || PixelSubdetector::PixelEndcap ){
+		if( SubDet == PixelSubdetector::PixelBarrel || SubDet == PixelSubdetector::PixelEndcap ){
```

Comment:
regarding this particular line of the fix, together with the other two similar occurrences in the other classes in the same package, this actually alters the behavior of the code but thanks to the protection few lines below: https://github.com/cms-sw/cmssw/pull/19231/files#diff-5b549e5390591af6a4ae6ece2166765bR689,  there is no issue with previous versions of the calibrations.